ui/src/app/approval-list/approval-list.component.spec.ts

```diff
@@ -1,25 +1,66 @@
-import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import {getTestBed, TestBed, tick} from '@angular/core/testing';
+import {HttpClientTestingModule, HttpTestingController} from '@angular/common/http/testing';
+import {HttpClient, HttpXhrBackend} from '@angular/common/http';
 
+import {environment} from '@environments/environment';
 import { ApprovalListComponent } from './approval-list.component';
+import {BaseService} from '@services/base.service';
+import {LoadingService} from '@services/loading.service';
+
 
 describe('ApprovalListComponent', () => {
   let component: ApprovalListComponent;
-  let fixture: ComponentFixture<ApprovalListComponent>;
-
-  beforeEach(async(() => {
-    TestBed.configureTestingModule({
-      declarations: [ ApprovalListComponent ]
-    })
-    .compileComponents();
-  }));
+  const http: HttpClient = new HttpClient(new HttpXhrBackend({ build: () => new XMLHttpRequest() }));
+  const loadingService: LoadingService = new LoadingService();
 
   beforeEach(() => {
-    fixture = TestBed.createComponent(ApprovalListComponent);
-    component = fixture.componentInstance;
-    fixture.detectChanges();
+    component = new ApprovalListComponent(http, loadingService);
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
 });
+
+describe('ApprovalApiService', () => {
+  let injector: TestBed;
+  const http: HttpClient = new HttpClient(new HttpXhrBackend({ build: () => new XMLHttpRequest() }));
+  const loadingService: LoadingService = new LoadingService();
+  const endpoint = 'v1/account/approvals/approve';
+  // const service: BaseService = new BaseService('v1/account/approvals/approve', http, loadingService);
+  let httpMock: HttpTestingController;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule],
+    });
+    injector = getTestBed();
+    httpMock = injector.get(HttpTestingController);
+  });
+
+  afterEach(() => {
+    httpMock.verify();
+  });
+
+  it('should return an Observable<string>', () => {
+    const expectedResponse = 'Accounts created. Existing account NOT updated.';
```

Comment:
@tommckennon The point of testing here is not to test the django rest api.  It is to test that the component reacts/does things based on certain responses from the api.  I see nothing in here that actually tests the component.

Reply:
I follow. Thank you.